src/createServer.js

```diff
@@ -1,3 +1,52 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+const { validateRequest } = require('./helpers/errorMessages');
+const {
+  formatErrorResponse,
+  formatSuccessResponse,
+} = require('./helpers/responseFormatter');
+
+const createServer = () => {
+  return http.createServer((req, res) => {
+    const normalizedUrl = new URL(req.url, 'http://localhost:5700');
+    const params = new URLSearchParams(normalizedUrl.searchParams);
+    const text = normalizedUrl.pathname.slice(1);
+    const toCase = params.get('toCase');
+
+    res.setHeader('Content-Type', 'application/json');
+
+    const validationErrors = validateRequest(
+      text,
+      toCase ? toCase.toUpperCase() : null,
```

Comment:
Ensure `toCase` is not null before calling `toUpperCase()`. You can add a check to handle cases where `toCase` is not provided, to prevent potential runtime errors.